nexus/tests/integration_tests/disks.rs

```diff
@@ -822,6 +829,126 @@ async fn test_disk_reject_total_size_not_divisible_by_min_disk_size(
     );
 }
 
+async fn create_instance_with_disk(client: &ClientTestContext) {
+    create_instance_with(
+        &client,
+        ORG_NAME,
+        PROJECT_NAME,
+        INSTANCE_NAME,
+        &params::InstanceNetworkInterfaceAttachment::Default,
+        vec![params::InstanceDiskAttachment::Attach(
+            params::InstanceDiskAttach { name: DISK_NAME.parse().unwrap() },
+        )],
+    )
+    .await;
+}
+
+const ALL_METRICS: [&'static str; 6] =
```

Comment:
Same point about duplicating these. I'm not sure how it impacts the test, but you can get these by listing the timeseries using [`oximeter_db::Client::list_timeseries`](https://github.com/oxidecomputer/omicron/blob/b858ed729f1ae8393def124f20e76831cf20f442/oximeter/db/src/client.rs#L113). That's not the easiest interface to use for this particular case, but we could improve it to allow finding metrics for a particular target name or something.

It's not blocking for this PR, since it's probably a lot of work, but I'm suggesting that in general we ask the database(s) what metrics are available, rather than writing them out here.

Reply:
If the call to `oximeter_db::Client::list_timeseries` returns fewer metrics than I expect, wouldn't I want to return an error? If I rely on that call to determine which metrics are supported, I worry that the test could silently skip checking for metrics which we'd still expect to be working, simply by omitting them from the test setup.

Reply:
So one other thought occurred to me. It would be nice if, in the case where a crate depends on Crucible, we could just refer to the names of the types that `impl oximeter::Metric` directly. That is, if you want to look at the `read_bytes` metric, you'd do something like `crucible::mod::with::stats::ReadBytes::name()`. We don't depend on Crucible directly here, and I don't know if that's better than a string even if we _do_. But this might be an argument in favor of a kind of shared library of metric type definitions, to make sure the producer and consumers are pulling the same names. I've thought about that off and on since building `oximeter`, but never really fleshed it out.

Reply:
A shared library of metric types would be nice - that would also make it easier for us to emulate a producer in testing, without needing to depend on the entirety of `crucible`.

Reply:
I kinda started this in `oximeter-instruments`, which currently just has the HTTP request latency stuff used in Nexus. We could conceivably put pretty general-purpose things here, though it might lead to some tension around fields. E.g., we might expose a "bytes written" type for both Crucible disks and physical drives, but I imagine the fields would be different. That could be captured by the _target_ fields though....Food for thought, I guess.